dvc/updater.py

```diff
@@ -48,7 +48,35 @@ def _with_lock(self, func, action):
             logger.debug(msg.format(self.lock.lockfile, action))
 
     def check(self):
-        if os.getenv("CI") or env2bool("DVC_TEST") or PKG == "snap":
+        if os.getenv("CI") or env2bool("DVC_TEST"):
+            return
+
+        if PKG == "snap":
+            # Ideally need to check if SNAP_CHANNEL.startswith(self.current),
+            # where SNAP_CHANNEL is:
+            #   snap info dvc | grep -E '^tracking:' | awk '{print $2}'
+            # However a quicker way is to skip if a user-configurable variable
+            # `skip_update_check` is set
+            import subprocess
+
+            kwargs = {"close_fds": True}
+            if os.name == "nt":
+                kwargs["shell"] = True
+                cmd = []
```

Comment:
Not sure I get why we do anything at all on windows.

Reply:
ah good point - I just copy-pasted this from `stage.run`